lib/application.js

```diff
@@ -115,24 +115,9 @@ app.model = function(Model, config) {
   if (arguments.length > 1) {
     config = config || {};
     if (typeof Model === 'string') {
-      // create & attach the model - backwards compatibility
-
-      // create config for loopback.modelFromConfig
-      var modelConfig = extend({}, config);
-      modelConfig.options = extend({}, config.options);
-      modelConfig.name = Model;
-
-      // modeller does not understand `dataSource` option
-      delete modelConfig.dataSource;
-
-      Model = registry.createModel(modelConfig);
-
-      // delete config options already applied
-      ['relations', 'base', 'acls', 'hidden', 'methods'].forEach(function(prop) {
-        delete config[prop];
-        if (config.options) delete config.options[prop];
-      });
-      delete config.properties;
+      throw new Error('Invalid model type of string found: ' +
+        Model +
+        '. Provide a valid model object to attach.');
```

Comment:
@bajtos: That's what I struggled with to come up with a decent message. Thanks for the update.